python/shotgun_model/shotgun_query_model.py

```diff
@@ -318,122 +287,117 @@ def reset(self):
             "email support@shotgunsoftware.com."
         )
 
-    ############################################################################
-    # protected properties
-
-    def _get_cache_path(self):
+    def hasChildren(self, index):
         """
-        Returns the cache path on disk for this instance.
+        Returns True if parent has any children; otherwise returns False.
+
+        This is used for the staged loading of nodes in hierarchies.
+
+        :param index: The index of the item being tested.
+        :type index: :class:`~PySide.QtCore.QModelIndex`
         """
-        return self.__full_cache_path
+        if not index.isValid():
+            return super(ShotgunQueryModel, self).hasChildren(index)
+
+        item = self.itemFromIndex(index)
+
+        if not isinstance(item, ShotgunStandardItem):
+            return super(ShotgunQueryModel, self).hasChildren(index)
```

Comment:
i think you could inject other types of items into the model possibly. i'll add a comment. again, this is old code.